### ISSUE TITLE
References and Some Entities Not Highlighted

### ISSUE THREAD

**Issue by kvn-s** [maelvalais/madoko-vscode#1]
At first thank you for making it available VS Code. Too bad that Madoko doesn't seem to be used that much. For me it's the perfect mixture of Markdown and Latex... :)

My small feature requests:
* Currently references such as `[@name17]` or `[@name17, p. 123]` aren't highlighted making them hard to find.
* Valid entities such as `&name-figure;` aren't not highlighted either.
Could you please add support for it?

Also I noticed that the [online editor](https://www.madoko.net/editor.html) has a very nice code completion feature for sections, bibliography, entities, block elements etc. making it a really tough decision trying to use it completely local. Do you know if it can be easily ported over to VS Code too?

**Comment by maelvls** [comment]
Yes, too bad it is not that used... 😞  It is so great!!

To be fair, the only thing I did in the 'extension' was to copy the `madoko.JSON-tmLanguage` that was in https://github.com/koka-lang/madoko. Generally speaking, I find any textmate language file hard to understand/update (even in YAML/JSON), but I'll try to take a look!

An IDE-like support (like in the [madoko editor](https://www.madoko.net/editor.html)) would be awesome. Unfortunately, I am really 'light' on javascript/typescript/overall vscode extension coding.

The good news is the online madoko editor relies on Monaco, which is also the editor in VSCode. It might ease the process of porting the whole thing to the vscode extension... Actually, I don't really know.

If someone with more knowledge on VSCode extensions want to give it a try, that would be great!!

**Comment by maelvls** [comment]
Hi! I fixed both coloration problems.

![capture d ecran 2017-10-18 a 18 02 14](https://user-images.githubusercontent.com/2195781/31729167-8ed34186-b42e-11e7-8a27-31adeca27cda.png)
![capture d ecran 2017-10-18 a 18 06 25](https://user-images.githubusercontent.com/2195781/31729369-17f3eff6-b42f-11e7-9548-895993af80ed.png)

For now, the pages (after a comma) is not colored at all but `@reference1` is colored.

Took me a while to figure out the right way to edit the `madoko.JSON-tmLanguage` file! By the way, the names given in this tmLanguage file are quite unusual, makes it hard to debug/maintain. Also, I used the VSCode's **`Developer: Inspect TM scopes`** to debug:
![capture d ecran 2017-10-18 a 18 12 37](https://user-images.githubusercontent.com/2195781/31729758-416c87de-b430-11e7-8272-3ab93afd4430.png)
